docs/technical-design/endpoints.md

```diff
@@ -0,0 +1,213 @@
+
+### Defining the type
+
+Of course, we have to have a good idea of what data is available to us before we can define our return type. In the MC-Review app, the database is the source of all our data, so we can see what's available by inspecting the `schema.prisma` file, which describes the database structure.
+
+Examine the model for a `State`. To keep this example as simple as possible, let's return the name of the first state in our database.
+
+#### The GraphQL query
+
+_**We'll create a new `.graphql` file in `/app-graphql/src/queries/dataExport.graphql`**_.
+
+We need to give our operation a name; let's use `dataExport`. Our query might look like this.
+
+```graphql
+query dataExport {
+    dataExport {
+        name
+    }
+}
+```
+
+`query` is the _operation type_. It's used to retrieve data. (We could also have a `mutation`, to change data, or a `subscription`, which is like a websocket, in that it maintains a connection to keep piping changing data.)
+
+`dataExport` is the _operation name_.
+
+We're telling GraphQL that we want a query called dataExport, which returns a field called "name".
+
+#### The domain model
+
+We use Typescript in our app, so all the data flowing from GraphQL through the api layer to the front end must also have a type. We keep the types that are shared between the api layer and the front end in our domain models.
+
+_**Create a new domain model in `/app-api/src/domain-models/DataExport.ts`**_ Put the following type in it.
+
+```typescript
+export type DataExportType = {
+    name: string | undefined
+}
+```
+
+You can think of this as the mirror-image of the GraphQL query, but for the application code.
+
+### Retrieving from the database
+
+#### The Prisma query
+
+Now let's get the data that we want.
+
+_**Create a new file: `/app-api/src/postgres/dataExport.ts`**_. We use Prisma to query the database.
+
+```typescript
+import { PrismaClient } from '@prisma/client'
+import { DataExportType } from '../domain-models/DataExport'
+
+export async function dataExport(
+    client: PrismaClient
+): Promise<DataExportType | undefined> {
+    const result = await client.state.findFirst()
+    if (result === undefined) {
+        return undefined
+    }
+    return {
+        name: result?.name,
+    }
+}
+```
+
+Recall the `schema.prisma` file. We're saying, in the `State` model (`client.state`, in this snippet), get the first record. From that result, give us the `name` field, and we'll assign it to a key of `name` in the object that we return. (Note that our object key can be whatever we want (as long as it matches the types we just made). It doesn't have to match the property name from the result, but in many cases that will make the most sense.)
+
+Also note that here we need the DataExportType that we just created in the domain model.
+
+#### Adding our method to the store
```

Comment:
```suggestion
#### Adding our method to the Store

We handle the database actions through an exported "Store", rather than calling the methods directly.
```
Do you think this kind of change could be helpful? Not sure if it adds clarity or not.